nexus/reconfigurator/planning/src/example.rs

```diff
@@ -326,8 +339,27 @@ impl ExampleSystemBuilder {
         self
     }
 
+    /// Set the policy for a sled in the example system by index.
+    ///
+    /// Returns an error if `index >= nsleds`.
+    pub fn with_sled_policy(
+        mut self,
+        index: usize,
+        policy: SledPolicy,
+    ) -> anyhow::Result<Self> {
+        if index >= self.sled_settings.len() {
```

Comment:
Two comments, both pretty minor:

* Could maybe use `sled_settings.get_mut(index)` to avoid the manual length check?
* ~~Is it useful for this method to return an error (as opposed to panicking directly)? It consumes self, so the caller presumably can't do anything useful with the system they're trying to build if they get this wrong.~~ Scratch this one; saw how this is used elsewhere.

Reply:
done.